src/scores/continuous/standard_impl.py

```diff
@@ -64,9 +64,7 @@ def mse(
     squared = scores.functions.apply_weights(squared, weights)
 
     if preserve_dims or reduce_dims:
-        reduce_dims = scores.utils.gather_dimensions(
-            fcst.dims, obs.dims, reduce_dims=reduce_dims, preserve_dims=preserve_dims
-        )
+        reduce_dims = scores.utils.gather_dimensions(fcst, obs, reduce_dims=reduce_dims, preserve_dims=preserve_dims)
```

Comment:
I'm not convinced by moving from passing the list of dimensions to passing the full fcst and obs logic. Passing just the list of dimensions makes it clear than only the names are used and that the logic is not data dependent. It also means that the tests require the creation of mock fcst and obs objects which (to my reading) makes it harder to comprehend the tests. I find the old test code for gather_dimensions easy to read, whereas now a complex intepretation of the object is required to understand what is being tested for. If it's not too much work, I would prefer the unified method to utilise the collection of dimension names rather than the full object, and to restore the previous test setup for the examples and exceptions where that is possible.

Reply:
@tennlee Are you still thinking you'd like only dims passed in or does the next comment trump this one?

Reply:
I could be wrong, but I do think it's a better design to have only the dims passed in, and that is what I'd like if that's okay.